### DISCUSSION TITLE
[Bug]: ERR_REQUIRE_ESM Node.js v20.9.0

### DISCUSSION THREAD

**Discussion by jlecordier** [spatie/pixelmatch-php#14] [Q&A] [answered]
### What happened?

When using `$this->assertMatchesImageSnapshot($path);`, I get the error :

```
Error Output:
================
/Users/jlecordier/Infoclimat/modeles-php/vendor/spatie/pixelmatch-php/bin/pixelmatch.js:5
const pixelmatch = require('pixelmatch');
                   ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/jlecordier/Infoclimat/modeles-php/node_modules/pixelmatch/index.js from /Users/jlecordier/Infoclimat/modeles-php/vendor/spatie/pixelmatch-php/bin/pixelmatch.js not supported.
Instead change the require of index.js in /Users/jlecordier/Infoclimat/modeles-php/vendor/spatie/pixelmatch-php/bin/pixelmatch.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/jlecordier/Infoclimat/modeles-php/vendor/spatie/pixelmatch-php/bin/pixelmatch.js:5:20) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v20.9.0
```

### How to reproduce the bug

- Node.js v20.9.0
- "spatie/pixelmatch-php": "^1.1"
- "spatie/phpunit-snapshot-assertions": "^5.1"
- package.json :
```json
{
  "dependencies": {
    "pixelmatch": "^6.0.0"
  }
}
```

### Package Version

1.1.0

### PHP Version

8.3.2

### Which operating systems does with happen with?

macOS

### Notes

Thanks for your work.

To make it work, I had to `npm i` inside the `pixelmatch-php` folder.

If I try to install `pixelmatch` globally, I get the error `Error: Cannot find module 'pngjs'`, even if I install `pngjs` globally

**Comment by poldixd** [comment]
pixelmatch version 6 is now a ES module and can not be used with `require()` (https://github.com/spatie/pixelmatch-php/blob/main/bin/pixelmatch.js#L5).

Try to use pixelmatch version 5.

```
npm install pixelmatch@5
```

**Reply by poldixd** [reply]
See the changelog from pixelmatch version 6 https://github.com/mapbox/pixelmatch/releases/tag/v6.0.0

**Comment by onlime** [comment, marked as answer]
#25 should fix it and you can install latest pixelmatch v7.1.0 with `npm install --save-dev pixelmatch`.
